Dev5

YanaShcherba opened the pull request.

YanaShcherba committed:
1.Code refactoring was done
2.Deleting button was fixed

YanaShcherba committed:
1.Modified filter action
2.Inline functions are replaced by methods
3.Added form for adding the new contact(in progress)

YanaShcherba committed:
1.all action creators are now arrow functions
2.handlers for filter&deletion are added to List component, constructor has been removed
3.added routes to form for adding new contact/editing existing contact
4.added the Form component itself. For this moment the new contact can be added
4.1 but the Dropzone for the photo is not working yet
4.1 but there's no redirection back to the list yet

YanaShcherba committed:
1.Added edition of name and number within existing contact
2.Added redirecting back to list after submitting changes

YanaShcherba committed:
1.Fixed editing of the existing contact (no more empty strings instead of name or number while changing only one value of inputs)
2.As i understood, dropzone was a bit..ehm..odd(or unnecessary) for this case, so it was replaced by input type="file".
3.Some styles were added, but there is a trouble: i'm doing something wrong,it seems.i can apply styles only for tagnames,not classnames

maksimshenets commented on src/components/List.jsx:
```diff
@@ -1,43 +1,53 @@
-                </button>
+      <div className="list-container">
+        <input
+          styleName="form-control"
```
styleName? It cause an error

YanaShcherba replied:
whoops.i just tried it,nothing good happened,and i forgot to remove it. there was className as well

maksimshenets replied:
ok, work in the same pull please

YanaShcherba replied:
yey,figured out what was the problem)

YanaShcherba replied:
everything was fixed (i hope).

maksimshenets commented on src/components/List.jsx:
```diff
@@ -1,43 +1,53 @@
+          {this.props.contactList
+            .filter(list => list.name.indexOf(this.props.visibilityFilter) !== -1)
+            .map((list) => (
+              <li key={list.id}><img src={list.image} alt="" />
```
![image](https://user-images.githubusercontent.com/12950715/28407959-b3f07416-6d3e-11e7-8b51-802e744f59c0.png)

maksimshenets commented on src/reducers/listReducer.jsx:
```diff
@@ -0,0 +1,71 @@
+    },
+  ],
+  visibilityFilter: '',
+  edit: {
```
Why we need this field in store?

YanaShcherba committed:
fixed the problem with classes in css-modules

YanaShcherba committed:
1.Removed unnecessary field from the store
2.Adding images via url is available
3.Buttons containing links were removed, all styles were added to links themselves,so no more need to click precisely into text of the link

maksimshenets commented:
Looks like everything works correct. @YanaShcherba resolve conflicts and merge.

maksimshenets commented:
BTW, why cats so angry?=)

YanaShcherba committed:
Merge branch 'master' into dev5

YanaShcherba commented:
probably someone on lorempixel has some kind of problems with cats)
